src/widgets/ActivityGroups/ui/ActivityCard/index.tsx

```diff
@@ -101,20 +102,45 @@ export const ActivityCard = ({ activityListItem }: Props) => {
 
   const restartActivity = () => {
     onStartActivity(true);
-    Mixpanel.track(MixEvents.ActivityRestarted, { [MixProperties.AppletId]: context.applet.id });
+
+    const analyticsPayload: MixpanelPayload = {
+      [MixpanelProps.AppletId]: context.applet.id,
+      [MixpanelProps.ActivityId]: activityListItem.activityId,
+    };
+
+    if (isFlow) {
+      analyticsPayload[MixpanelProps.ActivityFlowId] = activityListItem.flowId;
+    }
+
+    Mixpanel.track(MixpanelEvents.ActivityRestarted, analyticsPayload);
   };
+
   const resumeActivity = () => {
     onStartActivity(false);
-    Mixpanel.track(MixEvents.ActivityResumed, { [MixProperties.AppletId]: context.applet.id });
+
+    const analyticsPayload: MixpanelPayload = {
+      [MixpanelProps.AppletId]: context.applet.id,
+      [MixpanelProps.ActivityId]: activityListItem.activityId,
+    };
+
+    if (isFlow) {
+      analyticsPayload[MixpanelProps.ActivityFlowId] = activityListItem.flowId;
+    }
+
+    Mixpanel.track(MixpanelEvents.ActivityResumed, {
+      [MixpanelProps.AppletId]: context.applet.id,
+    });
   };
 
+  // Start activity on mount if direct linking to this activity; pass `true` to onStartActivity to
+  // ensure it doesn't resume from a previous progress state.
   useOnceEffect(() => {
     if (
       context.startActivityOrFlow &&
       ((!isFlow && context.startActivityOrFlow === activityListItem.activityId) ||
         (isFlow && context.startActivityOrFlow === activityListItem.flowId))
     ) {
-      restartActivity();
+      onStartActivity(true);
```

Comment:
The real source to the mounting problem eludes me, but I wouldn't be surprised if it had to do with lack of memoization in a parent component because a lot of functions inside hooks are being wrapped in `useCalled`.

However, for now we can fix it by using a ref.


```typescript
const previouslyMounted = useRef(false);

// Other code...

useOnceEffect(() => {
    if (previouslyMounted.current) {
      return;
    } else {
      previouslyMounted.current = true;
    }
```

Reply:
In fact, it may be useful to put this ref inside the definition of `useOnceEffect` 🤔

Reply:
So, I've tried this approach and it doesn't seem to work on my end. `previouslyMounted.current` is always evaluating as `false` and is never getting set to `true`, which means something else is going on.

I've also discovered more nuance around this:
- If the current activity being "Taken Now" was in a state of partial completion, the `useOnceEffect` only runs once.
- It's only when there is no saved activiy progress that the `useOnceEffect` runs twice.

I'll keep investigating, but feel free to contribute ideas!

Reply:
Strict mode only runs useEffect twice in dev mode, so I decided to test the production build as well. I was able to reproduce the behaviour in both cases. I then updated the definition of `useOnceEffect` here: https://github.com/ChildMindInstitute/mindlogger-web-refactor/commit/b0c7e1c0453347b83f545c3824cb872a8b455404

I added a console log at the top of the `useOnceEffect` call, and another right above the call to `onStartActivity`

```typescript
useOnceEffect(() => {
  console.log(`Running ActivityCard.useOnceEffect (name: ${title})`);
  if (
    context.startActivityOrFlow &&
    ((!isFlow && context.startActivityOrFlow === activityListItem.activityId) ||
      (isFlow && context.startActivityOrFlow === activityListItem.flowId))
  ) {
    console.log(`Running onStartActivity (name: ${title})`);
    onStartActivity(true);
  }
});
```

And I got this output in each scenario

```txt
Running ActivityCard.useOnceEffect (name: Activity 1)
Running onStartActivity (name: Activity 1)
Running ActivityCard.useOnceEffect (name: Flow1)
Running ActivityCard.useOnceEffect (name: A/B Trails iPad)
Running ActivityCard.useOnceEffect (name: A/B Trails Mobile)
Running ActivityCard.useOnceEffect (name: Simple & Choice Reaction Time Task Builder)
Running ActivityCard.useOnceEffect (name: CST Gyroscope)
Running ActivityCard.useOnceEffect (name: CST Touch)
```

Give that a test and let me know if it still runs twice

Reply:
FWIW I've just been disabling `<StrictMode>` altogether in `Main.tsx` while troubleshooting this, so that shouldn't be a factor for me. I've made the changes exactly you outlined above.

So long as I keep the activity in a state of partial-completion, it does **not** run twice:

![CleanShot 2024-06-25 at 13 51 48@2x](https://github.com/ChildMindInstitute/mindlogger-web-refactor/assets/1282772/9c13c8cd-deee-4b31-a7c8-958154dfff72)

But if I make sure I've completed the activity first, then start Take Now, it still continues to run twice:

![CleanShot 2024-06-25 at 13 51 22@2x](https://github.com/ChildMindInstitute/mindlogger-web-refactor/assets/1282772/f4c1b711-c331-4ff8-af00-706c044ef6e7)

But on your end, with your change you're unable to reproduce the buggy behaviour for both scenarios?

Reply:
Another detail worth noting – I'm not logging out after completing the activity. It only happens if I have an active Web App login session, and I'm starting Take Now for an activity in a not-in-progress state.